### DISCUSSION TITLE
Some strange issues with updates

### DISCUSSION THREAD

**Discussion by Timmiej93** [devbis/z03mmc#24] [General]
I have 8 thermometers scattered around the house. Some are near receivers/repeaters, others aren't. For those not near receivers, I expect updates to be infrequent, especially if they're indoors and temperatures don't change much (assuming the thermometers send out an update periodically and when large changes in values are detected).

There's a couple of weird ones though:
1. I have 2 sensors that are extremely close (unobstructed, 1 and 2m respectively) to the main receiver, that haven't been updated in 4 hours now. Other sensors, also ones that are further away from any receivers have updated just fine. What could be causing this? Battery levels are at 35% and 44%, would that be too low?
2. I have one sensor that's very close to a repeater (that I know works just fine) that also hasn't updated in 4 hours. For this specific sensor it could be that T and H haven't changed in that period, but I doubt it'd be exactly the same for 4 hours. 
3. I have 2 sensors inside vacuum sealed bags that contain filament and desiccant, but both humidity levels aren't updating, despite the temperature updating quite regularly. I checked on the display, both were displaying around 10% RH while in HA it's still showing around 50%, which was last updated roughly 4h ago.

The 4 hour pattern here most likely has something to do with me messing around with the sensors today. Some of the ones mentioned above are new, so I flashed them near the main receiver and then moved them to a repeater area, others I took the battery out of to "reset" them, since they were showing as unavailable this morning.

Before I moved to this firmware, I used [atc1441's firmware](https://github.com/atc1441/ATC_MiThermometer) with ESPHome. I moved away from that because I didn't like that you have to hard-code a thermometer to a specific receiver, so I moved to [pvvx's firmware](https://github.com/pvvx/ATC_MiThermometer) when I saw it supported BTHome. Unfortunately, about half of my devices just became unavailable after a while, despite being in the same location as before with the same (updated) receivers. Since I'm a big fan of Zigbee, especially with the size and shape of my house and my good experiences with it, I was hoping this firmware would solve everything, but it's looking like quite a few of my thermometers are gonna be unavailable again soon. Does anyone have any suggestions? Does Zigbee still do periodic network "repairs", or should that be more instantaneously these days?

**Comment by devbis** [comment]
Hi! Thanks for the report. Some people also reported about unavailable sensors after some time. I don't have a good strategy at the moment to catch this error, trying to figure out the culprit, but still no luck. 

Will post update if I find a solution for it.

**Comment by devbis** [comment]
https://github.com/devbis/z03mmc/issues/10#issuecomment-1783790084

Could you please try this firmware with sensors that become unavailable and check is the problem is solved ?

**Reply by Timmiej93** [reply]
I got a notification for the other comment as well ;). I'll first give 1.0.4 a try, after I figure out how to update ZigBee devices. If that doesn't help, I'll try the fw you linked and report back.

**Reply by Timmiej93** [reply]
Quick follow up question: Do we need to flash the [ATC_ota_40000.bin](https://github.com/devbis/z03mmc/releases/download/1.0.4/ATC_ota_40000.bin) too when updating Zigbee firmware? I assume not, but I'd rather ask than assume.

**Reply by devbis** [reply]
If you're on zigbee already, you can use zigbee OTA or uart upgrade. In both cases you don't need to flash intermediate firmware

**Reply by Timmiej93** [reply]
Thanks for the reply. Any tips for a specific OTA Zigbee method? I never updated any Zigbee device before. My initial searches all result in quite convoluted methods for updating.

**Reply by devbis** [reply]
https://github.com/devbis/z03mmc/issues/14 here is a description for local updates if you're using zha and here is a way to update from Zigbee2mqtt https://www.zigbee2mqtt.io/guide/usage/ota_updates.html#local-ota-index-and-firmware-files

**Reply by Timmiej93** [reply]
So I tried updating with ZHA, following the instructions you and sneakythr0ws provided, but I'm getting the "No OTA image is available" error.

I have the following folder, `config/zigpy_ota` containing the `1141-0203-10043001-z03mmc.zigbee` file. In my `configuration.yaml` file is the following:
```
zha:
  enable_quirks: True
  custom_quirks_path: custom_zha_quirks
  zigpy_config:
    ota:
      otau_directory: ./zigpy_ota
      ikea_provider: true
```

I tried all path variants I could think off, reloading the yaml inbetween, but it keeps saying there's no OTA image available. Any idea what could cause that?

Complete error:
```
2023-10-29 14:48:15.607 DEBUG (MainThread) [zigpy.zcl] [0x6CFB:1:0x0019] OTA query_next_image handler for 'Xiaomi LYWSD03MMC': field_control=0, manufacturer_id=4417, image_type=515, current_file_version=268578817, hardware_version=None, model='LYWSD03MMC'
2023-10-29 14:48:15.609 DEBUG (MainThread) [zigpy.zcl] [0x6CFB:1:0x0019] No OTA image is available
```

**Reply by devbis** [reply]
Unfortunately, I didn't have a change to check ZHA OTA. You'd better ask in https://github.com/devbis/z03mmc/issues/14  for instructions.

**Reply by Timmiej93** [reply]
Just to keep anyone tracking this thread, but not https://github.com/devbis/z03mmc/issues/14 up to date: Updating through ZHA is now working for me. ~~I _think_ it requires both the `.zigbee` and the `z03mmc.bin` files in the ota update folder.~~ ([source](https://github.com/devbis/z03mmc/issues/14#issuecomment-1784153790))

As for connectivity: I'll report back in a few days about that.

**Reply by Timmiej93** [reply]
Update on the connectivity on 1.0.4:
As mentioned before, I have 8 thermometers.
- 4 seem to be working perfectly fine
- 3 are completely unavailable, despite all being within a few meters from the main controller
- 1 is reporting temperature just fine, but humidity hasn't updated in 15 hours. Even when placing it on my PC exhaust, where the RH drops significantly, it doesn't update the RH.

I'm gonna give the custom fw you linked a try tomorrow or the day after. I see you released 1.0.5 today, did that change anything?

**Reply by Timmiej93** [reply]
Update:
I've been trying various versions on multiple devices, and unfortunately I haven't been able to get reliable readings on any of them. Some devices freeze, some update the screen correctly but only send either T or H to HA, and others don't send anything.

For now I'm going back to BT, to see if I can get that to work reliably.
